doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1671 @@
+2. TeamMgr to update the APP_DB LAG_TABLE with the new system-id-mac.
+3. TeamMgr to program the kernel with the new MAC address for the PortChannel interface and update the STATE_DB LAG_TABLE with the new system-id-mac address.
+
+<a id="333-Zebra"></a>
+### 3.3.3 Zebra
+Zebra changes are required to send ES membership and DF election information to Fpmsyncd. Existing socket for zebra-Fpmsyncd communication is leveraged. For sending split-horizon and DF information to Fpmsyncd, Trafic Control (TC) update messages will be used. Fpmsyncd will process raw netlink messages instead of using libnl3 APIs for decoding these messages.
+
+<a id="334-Fpmsyncd"></a>
+### 3.3.4 Fpmsyncd
+Fpmsyncd changes are required to handle ES membership and DF election information from zebra and write EVPN_SPLIT_HORIZON_TABLE and EVPN_DF_TABLE APP-DB entries.
+
+<a id="335-Fdbsyncd"></a>
+### 3.3.5 Fdbsyncd
+Fdbsyncd changes are required to handle the following:
+1. Handling of L2-NHG netlink notifications and dump from kernel.
+2. Programming of L2_NEXTHOP_GROUP_TABLE APP-DB entries based on kernel L2-NHG events.
+3. Programming of VXLAN_FDB_TABLE APP-DB entries with L2-NHID.
+4. Warm-reboot reconciliation of L2_NEXTHOP_GROUP_TABLE and VXLAN_FDB_TABLE similar to ROUTE_TABLE/NEXTHOP_GROUP_TABLE reconciliation in Fpmsyncd.
+
+<a id="336-Vxlanorch"></a>
+### 3.3.6 Vxlanorch
+- Require changes to maintain reference count of destination IP per L2 NHG.
+
+<a id="337-Fdborch"></a>
+### 3.3.7 Fdborch
+- FdbOrch is enhanced to process EVPN MH remote MAC updates via existing VXLAN_FDB_TABLE.
+- FdbOrch is extended to process new field "ifname" from VXLAN_FDB_TABLE entries.
+- FdbOrch is extended to process new field "type" for possible local/remote from VXLAN_FDB_TABLE entries.
+- FdbOrch is store and update MAC entries per ES associated access interface. This is to update the MAC entries when the associated access interface is going up/down.
+- FdbOrch is interact with EvpnMhOrch to check if the given access interface is associated with ES or not, fetch the L2-NHID associated with access interface, etc.
+- FdbOrch is extended to process FDB entries on local interface down and program then pointing to backup nexthop group using EVPN_ES_BACKUP_NHG_TABLE.
+
+Fdborch to process the local interface down and up in a similar fashion as for MCLAG.
+
+<a id="338-L2nhgorch"></a>
+### 3.3.8 L2nhgorch
+- New Orchagent class being introduced.
+- Subscription to different tables
+  - APP DB - L2_NEXTHOP_GROUP, L2_NEXTHOP_GROUP_MEMBER
+  - CONFIG DB - EVPN_ES_INTERFACE.
+- Subscription to L2_NEXTHOP_GROUP
+  - Creates/deletes the L2_ECMP_GROUP ASIC DB table.
+  - Interacts with PortsOrch to create a bridge-port object of type L2_ECMP_GROUP.
+- Subscription to L2_NEXTHOP_GROUP_MEMBER
+  - Creates/deletes the L2_ECMP_GROUP_MEMBER ASIC DB tables entries on the L2_NEXTHOP_GROUP APP DB table updates.
```

Comment:
The "DB Changes" and "Impact to existing modules" section conflicts with the L2_NEXTHOP_GROUP schema in APP_DB (the latter mentions a separate table for members). We propose that the one in "DB Changes" section is to be chosen as the correct approach and "L2nhgorch" subsection is to be rewritten accordingly.

Reply:
Text was updated. You may want to re-review this section.